icon-exclaim/icon-mpich/Dockerfile

```diff
@@ -1,38 +1,39 @@
 # syntax=docker/dockerfile:1.2
-FROM c2sm/nvhpc:21.3-devel-cuda_multi-ubuntu20.04-icon-mpich-dependencies
-ENV SPACK_PREFIX=/root/c2sm-spack
-ENV SPACK_SYSTEM_CONFIG_PATH=/root/c2sm-spack/sysconfigs/unknown
-ENV SPACK_ROOT=/root/c2sm-spack/spack
-
-RUN --mount=type=ssh spack install --fail-fast icon@exclaim-master%nvhpc +mpi ^netlib-lapack ~external-blas
-
-# dump paths to file
-RUN echo $(spack find --paths icon | grep -o -P "/root/c2sm-spack/spack/opt/.*")/bin/icon > /opt/icon
-
-# dump spack-env to file
-RUN echo $(spack load --sh icon) > /opt/spack-env
+FROM c2sm/nvhpc:21.3-devel-cuda_multi-ubuntu20.04-icon-mpich-dependencies as builder
+ENV ICON_SPEC="icon@container_build%nvhpc +mpi ^netlib-lapack ~external-blas"
+
+RUN --mount=type=ssh \
```

Comment:
Cool you made it work using environments :-)

Reply:
All the credit goes to Theo.